### PR TITLE
Lazy load registrant answers

### PR DESCRIPTION
**Pull request by adammeyer**
Lazy loading of registrant answers, get answers only when needed.

### CODE REVIEW

File: app/scripts/services/RegistrationCache.js [hunk 1]
```diff
@@ -80,17 +80,19 @@ angular.module('confRegistrationWebApp')
...
       var defer = $q.defer();
+      $rootScope.loadingMsg = 'Loading Registrations';
 
-      checkCache('conferences/' + conferenceId + '/registrations', function (registrations) {
-        angular.forEach(registrations, function (registration) {
-          update(path(registration.id), registration);
-        });
+      var blocksStr = '';
+      if(!_.isEmpty(blocks)){
+        blocksStr = '?block=' + blocks.join('&block=');
```
Comment (ryancarlson) [inline app/scripts/services/RegistrationCache.js]: I assume this function takes each element in the array and returns it as one string with `&block=` between each?

Comment (adammeyer) [inline app/scripts/services/RegistrationCache.js]: Correct.

File: app/views/eventRegistrations.html [hunk 2]
```diff
@@ -11,47 +11,44 @@
...
-                      {{block.title}}
-                    </a>
-                  </li>
-                </ul>
+            <ul class="dropdown-menu dropdown-menu-right reg-views-dropdown" ng-show="columnsDropdownToggle">
+              <li ng-repeat="block in blocks" ng-if="block.profileType !== 'NAME' && block.profileType !== 'EMAIL'">
+                <a href="" ng-click="toggleColumn($index)">
+                  <i class="fa fa-check-square" ng-show="block.visible"></i>
+                  <i class="fa fa-square-o" ng-show="!block.visible"></i>
+                  {{block.exportFieldTitle || block.title}}
```
Comment (ryancarlson) [inline app/views/eventRegistrations.html]: nice touch here!

File: app/scripts/controllers/exportData.js [hunk 3]
```diff
@@ -1,13 +1,13 @@
 'use strict';
 
 angular.module('confRegistrationWebApp')
-  .controller('exportDataModal', function ($scope, $modalInstance, conference, hasCost) {
+  .controller('exportDataModal', function ($scope, $modalInstance, $cookies, conference, apiUrl, hasCost) {
     $scope.conference = conference;
+    $scope.apiUrl = apiUrl;
+    $scope.authTocken = $cookies.crsToken;
```
Comment (ryancarlson) [inline app/scripts/controllers/exportData.js]: authTocken?

**Comment by ryancarlson** [comment]
Looks good if you can just clean up the authTocken reference :)

**Pull request merged**